### PR TITLE
Delete shape

### PR DESCRIPTION
**Pull request by levibyte**

### CODE REVIEW

File: src/commands/delete_command.hpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	virtual std::string get_name() override
+	{
+		return "dicmdDeleteShape";
+	}
+
+	virtual void execute() override
+	{
+		// find object from RQ
+		//FIXME need only one init during shape creation
+		RegionQuery& rq = RegionQuery::getInstance();
```
Comment (levibyte) [inline src/commands/delete_command.hpp]: Will be done via merge from viewport_operations.
Adding from tree will be done from directive shape creation commands on successful commit.

File: src/commands/delete_command.hpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+class dicmdDeleteObj : public DirectCommandBase
+{
+public:
+	dicmdDeleteObj(IObjectPoolPtr ptr, QPoint pos)
+		: m_workingSet(std::dynamic_pointer_cast<WorkingSet>(ptr))
+	{
+		add_option("-point",new PointCommandOptionValue(pos));
+	}
+
+	dicmdDeleteObj(IObjectPoolPtr ptr)
```
Comment (levibyte) [inline src/commands/delete_command.hpp]: We need at least one empty constructor for log-replaying to be working.
LogReply-er will create "empty" object and then add add_arg according to options from text files

File: src/commands/delete_command.hpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	InteractiveDeleteAction(IObjectPoolPtr ptr)
+		: m_workingSet(std::dynamic_pointer_cast<WorkingSet>(ptr))
+	{}
+
+	virtual std::string get_name() override
+	{
+		return "incmdDeleteShape";
+	}
+
+	virtual void handle_mouse_click(int x, int y) override
```
Comment (levibyte) [inline src/commands/delete_command.hpp]: Actually we don't need it. To grab last mouse position there is 
QPoint InteractiveCommandBase::get_last_point()

File: src/core/stackdump.hpp [hunk 4]
```diff
@@ -29,56 +29,57 @@
...
-#include <dlfcn.h>    // for dladdr
-#include <cxxabi.h>   // for __cxa_demangle
+//#include <execinfo.h> // for backtrace
+//#include <dlfcn.h>    // for dladdr
+//#include <cxxabi.h>   // for __cxa_demangle
 
 #include <cstdio>
 #include <cstdlib>
 #include <string>
 #include <sstream>
```
Comment (levibyte) [inline src/core/stackdump.hpp]: #ifdef OS_LINUX :) 
?

File: src/core/stackdump.hpp [hunk 5]
```diff
@@ -29,56 +29,57 @@
...
 
-#include <execinfo.h>
-#include <stdlib.h>
-#include <unistd.h>
-
+//#include <execinfo.h>
```
Comment (levibyte) [inline src/core/stackdump.hpp]: #ifdef OS_LINUX :)
?

File: src/core/working_set.cpp [hunk 6]
```diff
@@ -46,3 +46,12 @@ void WorkingSet::dumpToFile(const std::string& fname)
     file.flush();
     file.close();
 }
+
+void WorkingSet::deleteShape(IShape* obj)
```
Comment (levibyte) [inline src/core/working_set.cpp]: verchapes :))))))))))))))))))

File: src/core/working_set.hpp [hunk 7]
```diff
@@ -18,7 +18,8 @@ class WorkingSet : public IObjectPool
 	virtual void addObject(IShape*) override;
 	virtual ~WorkingSet() {}
 	virtual std::string getName() override;
-        virtual void dumpToFile(const std::string&);
+	virtual void dumpToFile(const std::string&);
+	void deleteShape(IShape*);
```
Comment (levibyte) [inline src/core/working_set.hpp]: may be removeObject?

File: src/gui/canvas.cpp [hunk 8]
```diff
@@ -71,7 +72,8 @@ canvas::canvas(QWidget* p)
     cm->register_command(new InteractiveDesAction<NEW>(m_working_set));   
     cm->register_command(new dicmdDesignSave(m_working_set));
     cm->register_command(new dicmdDesignLoad(m_working_set));
-        
+	cm->register_command(new InteractiveDeleteAction(m_working_set));
+	cm->register_command(new dicmdDeleteObj(m_working_set));
```
Comment (levibyte) [inline src/gui/canvas.cpp]: Also add dicmdDeleteObj in command_manager.cpp registry so log can be replied ))

File: src/commands/delete_command.hpp (hunk 2 above)
Comment (nagaina) [reply src/commands/delete_command.hpp]: dicmdCreateObj doesn't have default constructor. How does it work?

File: src/gui/canvas.cpp (hunk 8 above)
Comment (nagaina) [reply src/gui/canvas.cpp]: Can't find dicmdCreateObj in command_manager.cpp

**Pull request merged**